proposals/22-04-localize-command.md

```diff
@@ -0,0 +1,335 @@
+  - https://raw.githubusercontent.com/kubernetes-sigs/kustomize/v1.0.6/examples/helloWorld/configMap.yaml
+```
+
+I get an error from `kustomize build` in the pipeline because my configurations have remote references, but my CI/CD
+pipeline does not have external network access.
+
+Fortunately, I remember that I can run `kustomize localize` on `example/overlay` on my local machine. I can then upload
+the localized directory to my company’s internal package management site for the CI/CD pipeline to pull and build
+instead. I run
+`kustomize localize example/overlay example/localized-overlay`, but I get the following warning:
+
+```
+$ kustomize localize example/overlay example/localized-overlay
+Warning: File example/overlay/kustomization.yaml refers to ../base on line 2. This reference is outside of scope: 
+         example/overlay. kustomize localize will skip this path.
+```
+
+because I forgot that `kustomize localize` can only process local references to files within
+`scope`. Therefore, the command could not copy `example/base` to `example/localized-overlay`or localize the remote
+references in `example/base`. The resulting file structure is as follows:
+
+```
+└── example
+    ├── overlay
+    │   └── kustomization.yaml
+    ├── localized-overlay
+    │   ├── kustomization.yaml
+    │   └── localized-files
+    │       └── github.com
+    │           └── kubernetes-sigs
+    │               └── kustomize
+    │                   └── v1.0.6
+    │                       └── examples
+    │                           └── helloWorld
+    │                               └── deployment.yaml
+    └── base
+        └── kustomization.yaml
+```
+
+`kustomize build example/localized-overlay` will still run correctly outside the CI/CD pipeline because I chose to
+place `example/localized-overlay` in the same directory as `example/overlay`. As a result, relative paths,
+namely `../base`, in `example/localized-overlay` will point to the same files as their counterparts in `example/overlay`
+. However, `kustomize build example/localized-overlay` will still require network access to run.
+
+#### Story 2
+
+I am back again from **Story 1**, but this time ready to localize `base` too, instead of just the `overlay` directory.
+To localize both, I set my `scope` argument to `example` to make a copy of both directories. My setup still looks like
+that at the end of **Story 1**:
+
+```
+└── example
+    ├── overlay
+    │   └── kustomization.yaml
+    ├── localized-overlay
+    │   ├── kustomization.yaml
+    │   └── localized-files
+    │       └── github.com
+    │           └── kubernetes-sigs
+    │               └── kustomize
+    │                   └── v1.0.6
+    │                       └── examples
+    │                           └── helloWorld
+    │                               └── deployment.yaml
+    └── base
+        └── kustomization.yaml
+```
+
+After I run `kustomize localize example/overlay example new-space/localized-example`, I get the
```

Comment:
lol I wanted to demonstrate that we can make non-sibling `newDir`s. Since we're changing the outside of scope warning into an error, I can just get rid of the 1st user story and change this to a sibling directory.